Controller/RoleController.php

```diff
@@ -0,0 +1,227 @@
+            $this->addErrorMessage('role.error.role_not_found');
+
+            return $this->redirect(
+                $this->generateUrl('admin_roleList')
+            );
+        }
+
+        $roleData = (new RoleMapper())->mapToFormData($role);
+        $form = $this->createForm(
+            'ezrepoforms_role_update',
+            $roleData
+        );
+        $actionUrl = $this->generateUrl(
+            'admin_roleUpdate',
+            ['roleId' => $roleId]
+        );
+
+        // Synchronize form and data.
+        $form->handleRequest($request);
+        if ($form->isValid()) {
+            try {
+                $this->actionDispatcher->dispatchFormAction(
+                    $form,
+                    $roleData,
+                    $form->getClickedButton()->getName()
+                );
+            } catch (InvalidArgumentException $e) {
+                $this->addErrorMessage('role.error.name_exists');
+            } catch (UnauthorizedException $e) {
+                return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
+            }
+
+            if ($response = $this->actionDispatcher->getResponse()) {
+                return $response;
+            }
+
+            return $this->redirect($actionUrl);
+        }
+
+        // TODO: Just a temporary implementation of draft handling. To be done properly in follow-up: EZP-24701
+        $formView = $form->createView();
+        $roleName = $role->identifier;
+        if (preg_match('/^__new__[a-z0-9]{32}$/', $roleName) === 1) {
+            $roleName = 'role.name_new';
+            $formView->vars['role_input_value'] = '';
+        } else {
+            $formView->vars['role_input_value'] = $roleName;
+        }
+
+        return $this->render('eZPlatformUIBundle:Role:update_role.html.twig', [
+            'form' => $formView,
+            'action_url' => $actionUrl,
+            'role' => $role,
+            'role_name' => $roleName,
+        ]);
+    }
+
+    /**
+     * Deletes a role.
+     *
+     * @param int $roleId Role ID
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function deleteRoleAction($roleId)
+    {
+        try {
+            $role = $this->roleService->loadRole($roleId);
+        } catch (NotFoundException $e) {
+            $this->addErrorMessage('role.error.role_not_found');
+
+            return $this->redirect(
+                $this->generateUrl('admin_roleList')
+            );
+        }
+
+        try {
+            $this->roleService->deleterole($role);
+        } catch (UnauthorizedException $e) {
+            return $this->forward('eZPlatformUIBundle:Pjax:accessDenied');
+        }
+
+        return $this->redirect(
+            $this->generateUrl('admin_roleList')
+        );
+    }
+
+    /**
+     * Adds an error message to the flashbag.
+     *
+     * @param string $message Translatable error message
+     */
+    private function addErrorMessage($message)
```

Comment:
This should probably be moved out to abstract controller shared between them, and this should also probably allow dynamic input, like details on what specifically went wrong when short versions of that is possible to extract from exception.